### PR TITLE
[JAX FE] Support lax.argmax operation for JAX

### PR DESCRIPTION
**Pull request by halm-zenger**
### Details:
 - Support lax.argmax for JAX and create relevant layer test
 - 2 util improvements
   - Fix `num_inputs_check` not checking max inputs 
   - Better error message when param name not exist

### Tickets:
 - #26574

### CODE REVIEW

File: src/frontends/jax/src/utils.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ namespace jax {
 void num_inputs_check(const NodeContext& context, size_t min_inputs, size_t max_inputs) {
     auto inputs = context.inputs();
     FRONT_END_OP_CONVERSION_CHECK(inputs.size() >= min_inputs, "Got less inputs than expected");
+    FRONT_END_OP_CONVERSION_CHECK(inputs.size() <= max_inputs, "Got more inputs than expected");
```
Comment (halm-zenger) [inline src/frontends/jax/src/utils.cpp:19]: Not sure if I can fix this here, or maybe it's better to move it to a separate PR?

File: tests/layer_tests/jax_tests/test_argmax.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+from jax import numpy as jnp
+
+from jax_layer_test_class import JaxLayerTest
+
+
+def generate_shape_axis_pairs(input_shapes):
+    test_data = []
+    for shape in input_shapes:
+        rank = len(shape)
+        # Only [0, rank - 1] are valid axes for lax.argmax
```
Comment (halm-zenger) [inline tests/layer_tests/jax_tests/test_argmax.py:16]: Just note that [jax.numpy.argmax](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.argmax.html) can accept negative axis, but [jax.lax.argmax](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmax.html) cannot.

File: tests/layer_tests/jax_tests/test_argmax.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        [64],
+        [64, 16],
+        [48, 23, 54],
+        [2, 18, 32, 25],
+        [2, 18, 32, 25, 128],
+    ]
+    test_data = generate_shape_axis_pairs(input_shapes)
+
+    @pytest.mark.parametrize("params", test_data)
+    @pytest.mark.parametrize("index_dtype", [np.int32, np.int64])
```
Comment (halm-zenger) [inline tests/layer_tests/jax_tests/test_argmax.py:47]: I didn't handle `index_dtype` in translate_argmax but everything seems fine. Is it okay?

File: src/frontends/jax/src/utils.cpp (hunk 1 above)
Comment (rkazants) [reply src/frontends/jax/src/utils.cpp:19]: yes, let's do it.

File: tests/layer_tests/jax_tests/test_argmax.py (hunk 2 above)
Comment (rkazants) [reply tests/layer_tests/jax_tests/test_argmax.py:16]: [jax.numpy.argmax](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.argmax.html) is a wrapper and it is represented with a sub-graph of lax operations. For us, it is sufficient to check lax operation.

File: src/frontends/jax/src/utils.cpp (hunk 1 above)
Comment (halm-zenger) [reply src/frontends/jax/src/utils.cpp:19]: Thanks @rkazants Just to clarify, do you mean I should keep it here or I should move it to a separate PR?

Comment (halm-zenger) [reply src/frontends/jax/src/utils.cpp:19]: @rkazants it would be great if you  could please clarify this, thanks :)

Comment (rkazants) [reply src/frontends/jax/src/utils.cpp:19]: Yes, please add this check in this PR.

File: tests/layer_tests/jax_tests/test_argmax.py (hunk 3 above)
Comment (rkazants) [reply tests/layer_tests/jax_tests/test_argmax.py:47]: I think you need to check what index type is returned. Probably it makes sense to add `Convert` operation and parse `index_dtype` in the translator.

Comment (halm-zenger) [reply tests/layer_tests/jax_tests/test_argmax.py:47]: Thanks. Fixed in [9eddd5](https://github.com/openvinotoolkit/openvino/pull/26671/commits/9eddd531af7b05fd5a97aa517eb6459b008d2abb).

File: src/frontends/jax/include/openvino/frontend/jax/node_context.hpp [hunk 4]
```diff
@@ -101,6 +101,7 @@ class NodeContext : public frontend::NodeContext {
     }
 
     Output<Node> get_param(const std::string& name) const {
+        FRONT_END_GENERAL_CHECK(m_param_name_to_id.count(name), "No param id corresponding name: ", name, " exist.");
```
Comment (halm-zenger) [inline src/frontends/jax/include/openvino/frontend/jax/node_context.hpp:104]: If I change `context.const_named_param<int64_t>("axes")` (in `src/frontends/jax/src/op/argmax.cpp`) to `context.const_named_param<int64_t>("axis")`, master branch gave the following error message which is not very clear
```
    def convert(self, model: Union[Model, InputModel]) -> Model:
>       converted_model = super().convert(model)
E       openvino._pyopenvino.OpConversionFailure: Check 'is_conversion_successful' failed at src/frontends/jax/src/frontend.cpp:101:
E       FrontEnd API failed with OpConversionFailure:
E       Model wasn't fully converted. Failed operations detailed log:
E       -- argmax with a message:            
E       Exception happened during conversion of operation JaxFrameworkNode_9    
E       unordered_map::at       
E       Summary: 
E       -- Conversion is failed for: argmax 
```
This commit will provide the following better message:
```
    def convert(self, model: Union[Model, InputModel]) -> Model:
>       converted_model = super().convert(model)
E       openvino._pyopenvino.OpConversionFailure: Check 'is_conversion_successful' failed at src/frontends/jax/src/frontend.cpp:101:
E       FrontEnd API failed with OpConversionFailure:
E       Model wasn't fully converted. Failed operations detailed log:
E       -- argmax with a message:
E       Exception happened during conversion of operation JaxFrameworkNode_9
E       Check 'm_param_name_to_id.count(name)' failed at src/frontends/jax/include/openvino/frontend/jax/node_context.hpp:104:
E       FrontEnd API failed with GeneralFailure:
E       No param id corresponding name: axis exist.
E       
E       Summary:
E       -- Conversion is failed for: argmax
```
This will help with future developers' debugging.

Comment (rkazants) [inline src/frontends/jax/include/openvino/frontend/jax/node_context.hpp:104]:
```suggestion
        FRONT_END_GENERAL_CHECK(m_param_name_to_id.count(name), "No param id corresponding name exists: ", name);
```

File: tests/layer_tests/jax_tests/test_argmax.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        # Only [0, rank - 1] are valid axes for lax.argmax
+        valid_axes = range(0, rank)
+        test_data.extend([{'input_shape': shape, 'axis': axis}
+                         for axis in valid_axes])
+    return test_data
+
+
+class TestArgmax(JaxLayerTest):
+    def _prepare_input(self):
+        inp = jnp.array(np.random.rand(*self.input_shape).astype(np.float32))
```
Comment (rkazants) [inline tests/layer_tests/jax_tests/test_argmax.py:25]: please check with different input type. Also add a case when max element is duplicating in a tensor along given axis.

File: src/frontends/jax/src/op/argmax.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+OutputVector translate_argmax(const NodeContext& context) {
+    num_inputs_check(context, 1, 1);
+    Output<Node> input = context.get_input(0);
+    auto axis_val = context.const_named_param<int64_t>("axes");
+    auto axis = context.const_named_param<std::shared_ptr<v0::Constant>>("axes");
+    auto dtype = convert_dtype(context.const_named_param<int64_t>("index_dtype"));
+
+    auto k = std::make_shared<v0::Constant>(element::i64, Shape{}, 1);
+    auto topk = std::make_shared<v11::TopK>(input, k, axis_val, v11::TopK::Mode::MAX, v1::TopK::SortType::NONE);
```
Comment (rkazants) [inline src/frontends/jax/src/op/argmax.cpp:27]: what if maximum element is repeating in input tensor along given axis. Not sure that `v1::TopK::SortType::NONE` is suitable for it. Please check

File: tests/layer_tests/jax_tests/test_argmax.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        return jax_argmax, None, 'argmax'
+
+    input_shapes = [
+        [64],
+        [64, 16],
+        [48, 23, 54],
+        [2, 18, 32, 25],
+        [2, 18, 32, 25, 128],
+    ]
+    test_data = generate_shape_axis_pairs(input_shapes)
```
Comment (rkazants) [inline tests/layer_tests/jax_tests/test_argmax.py:44]: I think `generate_shape_axis_pairs` is not needed. Let define axis parameter value along with `input_shape` parameter. See `test_tf_ArgMinMax.py` test for tensorflow frontend.

Comment (halm-zenger) [reply tests/layer_tests/jax_tests/test_argmax.py]: Fixed in [c85dba](https://github.com/openvinotoolkit/openvino/pull/26671/commits/c85dba315525b6bd69475d3acf33b1598926ce08)

File: tests/layer_tests/jax_tests/test_argmax.py (hunk 5 above)
Comment (halm-zenger) [reply tests/layer_tests/jax_tests/test_argmax.py]: Fixed in [c85dba](https://github.com/openvinotoolkit/openvino/pull/26671/commits/c85dba315525b6bd69475d3acf33b1598926ce08)

File: src/frontends/jax/src/op/argmax.cpp (hunk 6 above)
Comment (halm-zenger) [reply src/frontends/jax/src/op/argmax.cpp]: Good point, previous implementation actually return index of last element, which is different than Jax. Industry standard is return index of first element, e.g. [numpy](https://numpy.org/doc/stable/reference/generated/numpy.argmax.html#numpy.argmax), [torch](https://pytorch.org/docs/main/generated/torch.max.html#torch.max), etc

Combination of `stable = true` and `v1::TopK::SortType::SORT_VALUES` will produce the same behaviour of Jax (return index of first max element)

Fixed in [c03797](https://github.com/openvinotoolkit/openvino/pull/26671/commits/c037970f38b95528453eaac9452915a198d2e23f)

File: tests/layer_tests/jax_tests/test_argmax.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                                                   ([48, 23, 54], 2),
+                                                   ([2, 18, 32, 25], 0),
+                                                   ([2, 18, 32, 25], 1),
+                                                   ([2, 18, 32, 25], 2),
+                                                   ([2, 18, 32, 25], 3),
+                                                   ([2, 18, 32, 25, 128], 0),
+                                                   ([2, 18, 32, 25, 128], 1),
+                                                   ([2, 18, 32, 25, 128], 2),
+                                                   ([2, 18, 32, 25, 128], 3),
+                                                   ([2, 18, 32, 25, 128], 4)])
```
Comment (rkazants) [inline tests/layer_tests/jax_tests/test_argmax.py:57]:
```suggestion
                                                   ([2, 18, 32, 25], 3)])
```

**Comment by rkazants** [comment]
build_jenkins

File: tests/layer_tests/jax_tests/test_argmax.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                                                   ([2, 18, 32, 25], 1),
+                                                   ([2, 18, 32, 25], 2),
+                                                   ([2, 18, 32, 25], 3)])
+    @pytest.mark.parametrize('input_type', [np.int8, np.uint8, np.int16, np.uint16,
+                                            np.int32, np.uint32, np.int64, np.uint64,
+                                            np.float16, np.float32, np.float64])
+    @pytest.mark.parametrize("index_dtype", [np.int32, np.int64])
+    @pytest.mark.parametrize("input_duplicate", [False, True])
+    @pytest.mark.nightly
+    @pytest.mark.precommit
```
Comment (rkazants) [inline tests/layer_tests/jax_tests/test_argmax.py:59]:
```suggestion
```

**Comment by rkazants** [comment]
build_jenkins

**Review by rkazants — APPROVED** [review]
